framework/src/actions/CreateProblemAction.C

```diff
@@ -71,25 +73,81 @@ CreateProblemAction::CreateProblemAction(InputParameters parameters)
 void
 CreateProblemAction::act()
 {
+  bool use_nonlinear = true;
+  bool use_eigenvalue = false;
+  if (_awh.hasActions("setup_executioner"))
+  {
+    CreateExecutionerAction * p = nullptr;
+    const auto & actions = _awh.getActionListByName("setup_executioner");
+    for (const auto & action : actions)
+    {
+      p = dynamic_cast<CreateExecutionerAction *>(action);
+      if (p)
+        break;
+    }
+    if (p)
+    {
+      auto & exparams = p->getObjectParams();
+      use_nonlinear = !(exparams.isParamValid("_eigen") && exparams.get<bool>("_eigen"));
+      use_eigenvalue =
+          (exparams.isParamValid("_use_eigen_value") && exparams.get<bool>("_use_eigen_value"));
+    }
+  }
+
+  if (_current_task == "meta_action")
+  {
+    // MOOSE set the default problem type on meta_action task, which allows other actions
+    // to reset problem type after through MooseApp::problemType().
+    if (!isParamValid("type"))
+    {
+      if (use_eigenvalue)
+        _app.problemType() = "EigenProblem";
+      else
+        _app.problemType() = "FEProblem";
+    }
+    else
+      _app.problemType() = getParam<std::string>("type");
+    return;
+  }
+
   // build the problem only if we have mesh
   if (_mesh.get() != NULL)
   {
+    // If users set type parameter in the input or the parameter is made valid by other actions,
+    // use it to set the problem type.
+    if (isParamValid("type"))
+      _app.problemType() = getParam<std::string>("type");
+
+    auto & type = _app.problemType();
+    InputParameters params = _factory.getValidParams(type);
+
+    if (_pars.have_parameter<bool>("skip_param_construction") &&
+        _pars.get<bool>("skip_param_construction"))
+      params.set<bool>("solve") = false;
```

Comment:
If somebody sees this statement, is it not clear why "skip_param_construction" implies "solve = false".  This was originally set in `createMinimalApp` so one knew the context for `solve = false`. The context is missing here...

Reply:
The problem is that `createMinimalApp` does not have access to the input parameters including `solve` for the FEProblem. `skip_param_construction` here is an indicator to the `CreateProblemAction` to skip extracting parameters for FEProblem in the input and set solve to false.

Reply:
Yes, we know it now. But, half a year from now, we will look at the code and be like "WTF?". The same with new people. It is not obvious that "skip_param_construction" and "solve" are related and how. So we would need a doco for the parameter when you correctly add it via `addParam` and possibly a comment here...

Reply:
Yes, agreed. A question is that do we want that parameter private or normal open to users?

Reply:
We can start with private and we can change it to public if we need to in future.